src/ImageSharp.Web/ImageSharpRequestAuthorizationUtilities.cs

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+#if !NETCOREAPP3_0_OR_GREATER
+using Microsoft.AspNetCore.Http.Internal;
+#endif
+
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Options;
+using SixLabors.ImageSharp.Web.Commands;
+using SixLabors.ImageSharp.Web.Middleware;
+using SixLabors.ImageSharp.Web.Processors;
+
+namespace SixLabors.ImageSharp.Web
+{
+    /// <summary>
+    /// Contains various helper methods for authorizing image requests.
+    /// </summary>
+    public sealed class ImageSharpRequestAuthorizationUtilities
```

Comment:
Shouldn't this be renamed to `RequestAuthorizationUtilities`, similar to `FormatUtilities`, `HMACUtilities`, etc?

Reply:
I wanted to make it more explicit. It’s for ImageSharp requests only.

Reply:
Yeah, but it's already in the `SixLabors.ImageSharp.Web` namespace and consumers can otherwise alias the class name if it would conflict with something else:
```c#
using ImageSharpRequestAuthorizationUtilities = SixLabors.ImageSharp.Web.RequestAuthorizationUtilities;
```

And after mentioning that, I acknowledge that you can also use that to do the reverse 😓 So it's more to align with the naming used by all the other utility classes...

Reply:
The naming is about about consumption intent. All the other types are designed to be used conceptionally within other ImageSharp.Web type implementations (processors generally) whereas this type is designed to be consumed from the users code only. It has to be explicit and unfortunately verbose.